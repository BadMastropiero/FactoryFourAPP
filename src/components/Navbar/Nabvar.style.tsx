import styled from 'styled-components';

export const StyledNavbar = styled.header`
  background-color: ${(props) => props.theme.colors.primary.base};
  color: ${(props) => props.theme.colors.primary.contrast};
  display: flex;
  align-items: center;
  justify-content: center;
`;

export const StyledNavbarContent = styled.nav`
  display: flex;
  align-items: center;
  justify-content: space-between;
  max-width: ${(props) => props.theme.layout.maxWidth};
  width: 100%;
  padding: ${(props) => [props.theme.layout.vGap, props.theme.layout.hGap].join(' ')};
`;
