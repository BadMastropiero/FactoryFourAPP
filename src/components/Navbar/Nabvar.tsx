import { NavbarProps } from './Nabvar.interface';
import { StyledNavbar, StyledNavbarContent } from './Nabvar.style';

// export default function Navbar({ title }: NavbarProps) {
//   return (
//     <StyledNavbar>
//       <StyledNavbarContent>
//         <h3>{title}</h3>
//       </StyledNavbarContent>
//     </StyledNavbar>
//   );
// }

const Navbar: React.FC<NavbarProps> = ({ title }) => {
  return (
    <StyledNavbar>
      <StyledNavbarContent>
        <h3>{title}</h3>
      </StyledNavbarContent>
    </StyledNavbar>
  );
};

export default Navbar;
