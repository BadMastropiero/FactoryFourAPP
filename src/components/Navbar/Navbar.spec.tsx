import '@testing-library/jest-dom';

import { render, screen } from '@testing-library/react';
import { ThemeProvider } from 'styled-components';
import { describe, expect, it } from 'vitest';

import { theme } from '../../styles/theme';
import Navbar from './Nabvar';

describe('Navbar', () => {
  it('renders correctly', () => {
    const { asFragment } = render(
      <ThemeProvider theme={theme}>
        <Navbar title="Test Navbar" />
      </ThemeProvider>,
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it('renders the title prop', () => {
    const title = 'Status Dashboard';
    render(
      <ThemeProvider theme={theme}>
        <Navbar title="Status Dashboard" />
      </ThemeProvider>,
    );
    expect(screen.getByText(title)).toBeInTheDocument();
  });
});
