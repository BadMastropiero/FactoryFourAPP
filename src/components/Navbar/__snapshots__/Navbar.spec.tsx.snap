// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Navbar > renders correctly 1`] = `
<DocumentFragment>
  <header
    class="sc-bdfCDU fjQoCB"
  >
    <nav
      class="sc-gsTDqH gdwkMO"
    >
      <h3>
        Test Navbar
      </h3>
    </nav>
  </header>
</DocumentFragment>
`;
